app/entrypoint.sh

```diff
@@ -67,7 +67,7 @@ if [ "$1" = 'mattermost' ]; then
     echo "Using existing config file" $MM_CONFIG
   fi
 
-  if [ ! -z "$DB_HOST" ]
+  if [ -z "$MM_SQLSETTINGS_DATASOURCE" ] && [ ! -f $MM_CONFIG ]
```

Comment:
Don't sure if it will works. If the file `MM_CONFIG` does not exists it will be created before in this script (see line 30), so I guess this condition will never be true.

Reply:
i tested all the cases here
1. with a mm config
2. without mm config but the MM_SQLSETTINGS_DATASOURCE set
3. without mm config and without MM_SQLSETTINGS_DATASOURCE set

Reply:
This is weird. If there is no configuration file it will be created by [this line](https://github.com/mattermost/mattermost-docker/blob/b57346b185e530974ce6ebcc6e8e9b1e01250ac1/app/entrypoint.sh#L36) so, at this point, a configuration file should always exists.

Reply:
can you try on your side?

Reply:
I'll do it later to check yes.

Reply:
thanks @pichouk

Reply:
I confirm on my side, after some test, that this condition is never `true` when using a "basic" Docker Compose install. Container is correctly working, but the lines under this condition are here for safety, to prevent Mattermost application to boot if the database is not reachable. We have 2 options :
- remove this check of the database and consider that if database is not available then application will not boot anyway
- fix this condition to match your case and the "basic" one

Reply:
Thanks for checking.
I agree to remove the check. This is performed by the app.

Agree to remove @pichouk

Reply:
Yes LGTM :+1:

Reply:
ok, removed PTAL